tests/mypy_test.py

```diff
@@ -252,6 +252,11 @@ def get_mypy_flags(args: TestConfig, temp_name: str, *, testing_stdlib: bool) ->
         "--custom-typeshed-dir",
         str(Path(__file__).parent.parent),
         "--strict",
+        # Ignoring "module is installed, but missing library stubs or py.typed marker"
+        # because external dependencies may not be marked as typed.
+        # pyright already flags non-typed external dependencies as warnings.
+        # TODO: This mutes ALL import errors, can we narrow it down?
+        "--ignore-missing-imports",
```

Comment:
I'm not sure there's much use for a stubs package having a dependency on a non-types library that doesn't have inline types. Stubs packages should keep dependencies to an absolute minimum, so that they pose minimal security risks and are easy to install. Since the sole purpose of a stubs package is to help with type checking, I'd rather we didn't allow adding dependencies that don't have inline types.

I also think adding this flag may be obscuring the fact that `mypy_test.py` is probably ignoring all packages installed in `site-packages` due to the `--no-site-packages` flag on line 251!

```suggestion
```

Reply:
The reason I was including non- `py.typed` packages, is because, if I'm not mistaken, pyright will still get the proper type (let's say a class). Although mypy does import everything as `Any` in those cases. I only found two cases of this:
1. `comtypes` : used by D3DShot, so arguably already on its way out. If another stub ever needs it, maybe consider also typing `comtypes`
2. `modulegraph`: used by PyInstaller. However PyInstaller currently vendors modulegraph, so we can just stub the necessary vendored files (which is already what I did).

What I was thinking on doing, is to add
```
[mypy-non_type_dependency]
ignore-missing-imports=true
```
to a config file (like `mypy_test` does) so that only those specific non-type imports are muted.

That being said... this counter-argument is pretty weak and a niche/edge case probably not worth supporting and adding complexity. So I'm with you on this one.

Reply:
That's a good point, actually, pyright has the `useLibraryCodeForTypes` option. I don't think mypy has anything similar, though, so I still think we should be pretty wary about adding "non-`py.typed` non-types dependencies". If we _do_ decide to allow them (which I think would probably require a bit more discussion first, but I'm open to it if it'll help pyright users 🙂), I think it might actually _still_ be better to have to use a `type: ignore[import]` each time the library is imported. That way, there's a very visual reminder that this library is "a bit different" and should be used with caution inside typeshed.